src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawerAdapter.java

```diff
@@ -1,20 +1,36 @@
 package edu.kis.powp.jobs2d.drivers.adapter;
 
+import com.google.inject.Inject;
 import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+import javax.sound.sampled.Line;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawerAdapter implements Job2dDriver {
 	private int startX = 0, startY = 0;
+	private ILine line = LineFactory.getBasicLine();
+
+	@Inject
+	DrawerFeature drawerFeature;
+
 
-	public MyAdapter() {
+	public DrawerAdapter() {
 		super();
 	}
 
+	public void choose_line(int choose)
+	{
+		if(choose==1) line = LineFactory.getBasicLine();
```

Comment:
This is solution to 3.3.2 (not 3.3.3)
unnecessary direct coupling with LineFactory (line setter would be better)